mmv1/templates/terraform/pre_update/alloydb_instance.go.erb

```diff
@@ -1,3 +1,10 @@
+// Avoid modifying instance type. No use case for this
+// Secondary instances can not be created in primary cluster and vice versa.
+// Default change is destroy before create, so modifcation from primary <-> secondary will only delete the instance with failure to create the instance.
+if d.HasChange("instance_type") {
```

Comment:
This should be done by marking the field as immutable- we can use a `deletion_policy` to require users to explicitly specify the resource should be abandoned instead.